Mathlib/CategoryTheory/Idempotents/KaroubiKaroubi.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+
+! This file was ported from Lean 3 source module category_theory.idempotents.karoubi_karoubi
+! leanprover-community/mathlib commit 31019c2504b17f85af7e0577585fad996935a317
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Idempotents.Karoubi
+
+/-!
+# Idempotence of the Karoubi envelope
+
+In this file, we construct the equivalence of categories
+`KaroubiKaroubi.equivalence C : Karoubi C ≌ Karoubi (Karoubi C)` for any category `C`.
+
+-/
+
+
+open CategoryTheory.Category
+
+open CategoryTheory.Idempotents.Karoubi
+
+namespace CategoryTheory
+
+namespace Idempotents
+
+namespace KaroubiKaroubi
+
+variable (C : Type _) [Category C]
+
+-- porting note: added to ease automation
+@[simp]
```

Comment:
Should this also be `reassoc`?

Reply:
Thanks. Yes, indeed it should. I have just added it.